libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1006,6 +1006,14 @@ linearize preds graph = do
 -- Computing specs
 ------------------------------------------------------------------------
 
+simplifySpec :: HasSpec fn a => Specification fn a -> Specification fn a
+simplifySpec = \case
+  SuspendedSpec x p -> computeSpecSimplified x (optimisePred p)
+  MemberSpec xs -> MemberSpec (nub xs)
+  ErrorSpec es -> ErrorSpec es
+  TypeSpec ts cant -> TypeSpec ts (nub cant)
+  TrueSpec -> TrueSpec
+
```

Comment:
From my observation of the changes.  This simplify spec is used as a view pattern in a few places. Is it used elsewhere?

Reply:
Basically nowhere else.